Source/SwiftLintBuiltInRules/Rules/Idiomatic/UnusedEnumeratedRule.swift

```diff
@@ -18,43 +18,177 @@ struct UnusedEnumeratedRule: Rule {
             Example("for (idx, _) in bar.enumerated().something() { }"),
             Example("for (idx, _) in bar.something() { }"),
             Example("for idx in bar.indices { }"),
-            Example("for (section, (event, _)) in data.enumerated() {}")
+            Example("for (section, (event, _)) in data.enumerated() {}"),
+            Example("list.enumerated().map { idx, elem in \"\\(idx): \\(elem)\" }"),
+            Example("list.enumerated().map { $0 + $1 }"),
+            Example("list.enumerated().something().map { _, elem in elem }"),
+            Example("list.map { ($0.offset, $0.element) }"),
+            Example("""
+            list.enumerated().map {
+                $1.enumerated().forEach { print($0, $1) }
+                return $0
+            }
+            """)
         ],
         triggeringExamples: [
             Example("for (↓_, foo) in bar.enumerated() { }"),
             Example("for (↓_, foo) in abc.bar.enumerated() { }"),
             Example("for (↓_, foo) in abc.something().enumerated() { }"),
-            Example("for (idx, ↓_) in bar.enumerated() { }")
+            Example("for (idx, ↓_) in bar.enumerated() { }"),
+            Example("list.enumerated().map { idx, ↓_ in idx }"),
+            Example("list.enumerated().map { ↓_, elem in elem }"),
+            Example("list.enumerated().forEach { print(↓$0) }"),
+            Example("list.enumerated().map { ↓$1 }"),
+            Example("""
+            list.enumerated().map {
+                $1.enumerated().forEach { print(↓$1) }
```

Comment:
> It feels a bit strange that the rule triggers on the identifiers (that are not missing) even though the rule is about the `enumerated()` method.

I know what you mean - it's not like the `_` case where it's obvious. I think of it as `$0` (or `$1`) being the thing that indicates there's a problem, as we didn't expect to see only one of `$0` or `$1`.

Reply:
I'd still expect the rule to trigger on the `enumerated()` call to be honest. Changing that shouldn't be a big deal, right?

Reply:
Yep, that should be possible.